library/std/src/sys/unix/fs.rs

```diff
@@ -1077,8 +1079,10 @@ impl File {
 
     pub fn set_times(&self, times: FileTimes) -> io::Result<()> {
         cfg_if::cfg_if! {
-            if #[cfg(target_os = "redox")] {
+            if #[cfg(any(target_os = "redox", target_os = "espidf"))] {
                 // Redox doesn't appear to support `UTIME_OMIT`.
+                // ESP-IDF does not support the `futimens` function.
+                // It would need a `utime`-based code-path in future.
```

Comment:
```suggestion
```

`utime` is filename-based, not fd-based. Does ESP-IDF have an fd-based API to use here?

In any case, let's not speculate on what path future additions might take here.

Reply:
Agreed and comment fixed accordingly.